### DISCUSSION TITLE
WCF client to connect to legaccy apps where should be configured?

### DISCUSSION THREAD

**Discussion by zinov** [jasontaylordev/CleanArchitecture#946] [Q&A]
if a WCF service is going to be consumed from the API project to connect to Legacy apps, where should I place that dependency? on the infrastructure project?

**Comment by DennisJansenDev** [comment]
> if a WCF service is going to be consumed from the API project to connect to Legacy apps, where should I place that dependency? on the infrastructure project? 

Yes, all external connectivity should reside in the infrastructure project. That means event buses, third party api's, database connections and also WCF services.

**Reply by zinov** [reply]
If I have a 3rd party APIs, like a rest API cal, why not create a QueryHandler at the application layer that handles that request rather then creating this at the infrastructure level?

Do you have any repo examples that I can inspect?
